test(coverage/typescript): Exclude more non-testable units

leaysgur opened the pull request:
Part of #11582 

The filter logic added in the previous PR was not complete, so I fixed it.
Also added more non-testable case.

~~For `parser_typescript.snap`, 2 "Expect Syntax Error" entries are newly reported, but they will be handled in the future task.~~

leaysgur commented:
> For parser_typescript.snap, 2 "Expect Syntax Error" entries are newly reported, but they will be handled in the future task.

Addressed this. 👉🏻  https://github.com/oxc-project/oxc/pull/12159/commits/398e2f516e9e7aadb9d32c589d365bdf77b6bb9e

leaysgur commented:
@Boshen PTAL when you have time. 👀

Boshen commented:
### Merge activity

* **Jul 15, 2:18 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
test(coverage/typescript): Exclude more non-testable units (#12159)

Part of #11582

The filter logic added in the previous PR was not complete, so I fixed it.
Also added more non-testable case.

~~For `parser_typescript.snap`, 2 "Expect Syntax Error" entries are newly reported, but they will be handled in the future task.~~
